weblate/formats/ttkit.py

```diff
@@ -1490,18 +1490,39 @@ class XWikiPropertiesFormat(PropertiesBaseFormat):
     autoload = ("*.properties",)
     new_translation = "\n"
 
+    # Ensure that not translated units are saved too as missing properties and
+    # comments are preserved as in the original source file.
     def save_content(self, handle):
         current_units = self.all_units
+        store_units = self.store.units
+
+        # We empty the store units since we want to control what we'll serialize
         self.store.units = []
-        # Ensure that not translated units are saved too as missing properties.
+
         for unit in current_units:
-            if unit.unit is None:
-                if not unit.has_content():
-                    unit.unit = unit.mainunit
+            # If the translation unit is missing and the current unit is not
+            # only about comment.
+            if unit.unit is None and unit.has_content():
+
+                # We first check if the unit has not been translated as part of a
+                # new language: in that case the unit is not linked yet.
+                found_store_unit = None
+                for store_unit in store_units:
+                    if unit.context == store_unit.name:
+                        found_store_unit = store_unit
```

Comment:
This will perform horribly, because you are iterating N*N times for N units.

Reply:
Yeah I know that's not great but I didn't find any better for doing this for now.. Note that the loop only occur if there's no pounit attached and it's not a comment. 
Clearly the bug is here: apparently when adding a new language the pounit are not properly attached to the units when I loop over them. It might be related to the fact that all_units is cached, so I'm getting an old version of the units, but when I looked for another way to perform this loop I didn't find it. 

Taking any advice to better handle this :)